src/util/support/dir_filenames.c

```diff
@@ -87,7 +87,7 @@ k5_dir_filenames(const char *dirname, char ***fnames_out)
         return ENOENT;
 
     do {
-        if (add_filename(&fnames, &n_fnames, &ffd.cFileName) != 0) {
+        if (add_filename(&fnames, &n_fnames, ffd.cFileName) != 0) {
```

Comment:
It looks like `WIN32_FIND_DATA.cFileName` is a `CHAR cFileName[PATH_MAX]`, and IIRC Unix compilers have `&` do the same thing as the implicit conversion from array to pointer type.  But the C spec doesn't really seem to support this, saying that the argument to the `&` operator must be "... or an lvalue that designates an object that is not a bit-field".  The array itself is not, IIUC, an lvalue...

Reply:
The two expressions have the same numeric value, so this bug is harmless, but the former expression has type `char (*)[PATH_MAX]`.  If the same code were compiled in the main build, we would see a warning like this one:

    warning: passing argument 1 of ‘fn’ from incompatible pointer type [-Wincompatible-pointer-types]
    note: expected ‘const char *’ but argument is of type ‘char (*)[100]’

Reply:
I don't see how to get that type from what n1256.pdf says, but I also think we should just merge the PR and not spend more time on this question :)

Reply:
Note that 6.3.2.1 says "A modifiable lvalue is an lvalue that does not have array type [...]".  So ffd.cFileName is an lvalue, just not a modifiable lvalue.

Reply:
Ah, thanks for the pointer (pun intended).